internal/infrastructure/kubernetes/proxy/testdata/daemonsets/override-labels-and-annotations.yaml

```diff
@@ -215,6 +215,21 @@ spec:
               typed_config:
                 "@type": type.googleapis.com/envoy.extensions.resource_monitors.downstream_connections.v3.DownstreamConnectionsConfig
                 max_active_downstream_connections: 50000
+            - name: "envoy.resource_monitors.fixed_heap"
```

Comment:
why these changes happen?

Reply:
It's set based on the memory limit by default, since memory limits weren't defined before they weren't being applied

Code here https://github.com/envoyproxy/gateway/blob/2a10d474fdfe2f3a9689266355fa6d89a042daac/internal/infrastructure/kubernetes/proxy/resource.go#L356-L369

Reply:
thanks for clarify